rbac/management/role/view.py

```diff
@@ -645,3 +646,15 @@ def delete_policies_if_no_role_attached(self, role):
         for policy in policies:
             if policy.roles.count() == 1:
                 policy.delete()
+
+    def restrict_role_create_update_if_system_role_exists(self, request):
+        """If a system role has the same display_name when a role is being created or updated restrict the operation."""
+        req_method = request.method
+        req_name = request.data.get("name")
+        public_tenant = Tenant.objects.get(tenant_name="public")
+        base_role_queryset = Role.objects.filter(display_name=req_name, tenant__in=[request.tenant, public_tenant])
+        req_system_role_exists = base_role_queryset.filter(system=True).exists()
+        if req_system_role_exists and req_method == "PUT":
```

Comment:
Due to the specifications, we have PUT and PATCH methods that are able to change role information.
Is PATCH also covered here?